unit-tests/func/options/test-color-options.py

```diff
@@ -0,0 +1,95 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#test:device L500*
+#test:device D400*
+
+import pyrealsense2 as rs
+from rspy import test
+
+color_options = [rs.option.backlight_compensation,
+                     rs.option.brightness,
+                     rs.option.contrast,
+                     rs.option.gamma,
+                     rs.option.hue,
+                     rs.option.saturation,
+                     rs.option.sharpness,
+                     rs.option.enable_auto_white_balance,
+                     rs.option.white_balance]
+
+color_metadata = [rs.frame_metadata_value.backlight_compensation,
+                     rs.frame_metadata_value.brightness,
+                     rs.frame_metadata_value.contrast,
+                     rs.frame_metadata_value.gamma,
+                     rs.frame_metadata_value.hue,
+                     rs.frame_metadata_value.saturation,
+                     rs.frame_metadata_value.sharpness,
+                     rs.frame_metadata_value.auto_white_balance_temperature,
+                     rs.frame_metadata_value.white_balance]
+
+def check_option_and_metadata_values(option, metadata, value_to_set, frame):
+    changed = color_sensor.get_option(option)
+    if changed != value_to_set:
+        print("failure in setting option: " + repr(option) + "with value: " + repr(value_to_set) +
+              "get_option returned " + repr(changed))
+        test.fail()
+    if frame.supports_frame_metadata(metadata):
+        changed_md = frame.get_frame_metadata(metadata)
+        if changed_md != value_to_set:
+            print("failure in setting option: " + repr(option) + "with value: " + repr(value_to_set) +
+                ", get_frame_metadata returned " + repr(changed_md))
+            test.fail()
+    else:
+        print("metadata " + repr(metadata) + " not supported")
+
+ctx = rs.context()
+dev = ctx.query_devices()[0]
+color_sensor = dev.first_color_sensor()
```

Comment:
Check the return value before dereferencing.
RGB is not available for SKUs such as SR306

Reply:
Done